src/harness/testcases/WINNF_FT_S_SSS_testcase.py

```diff
@@ -94,17 +98,52 @@ def generate_SSS_9_default_config(self, filename):
 
   @configurable_testcase(generate_SSS_9_default_config)
   def test_WINNF_FT_S_SSS_9(self, config_filename):
-    """Non-CBRS trust root signed certificate presented during registration.
+    """Non-CBRS trust root signed certificate presented by SAS Test Harness.
 
     Checks that SAS UUT response with fatal alert message.
     """
     config = loadConfig(config_filename)
     self.assertTlsHandshakeFailure(client_cert=config['sasCert'],
                                    client_key=config['sasKey'])
+  def generate_SSS_10_default_config(self, filename):
+    """Generates the WinnForum configuration for SSS_10. """
+    # Create the actual config for SAS cert/key path 
+
+    config = {
+        'sasCert': self.getCertFilename("wrong_type_sas.cert"),
+        'sasKey': self.getCertFilename("client.key")
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SSS_10_default_config)
+  def test_WINNF_FT_S_SSS_10(self, config_filename):
+    """Certificate of wrong type presented during Full Activity Dump.
+
+    Checks that SAS UUT response with fatal alert message.
+    """
+    config = loadConfig(config_filename)
+
+    # Reset the SAS UUT
+    self.SasReset()
+
+    # Notify SAS UUT about peer SAS
+    certificate_hash = getCertificateFingerprint(config['sasCert'])
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,\
+                                     'url': SAS_TEST_HARNESS_URL})
+    try:
+      self.assertTlsHandshakeFailure(client_cert=config['sasCert'],
+                                     client_key=config['sasKey'])
+    except AssertionError as e:
+      try:
+         self.TriggerFullActivityDumpAndWaitUntilComplete(config['sasCert'], config['sasKey'])
+         self.fail("Full Activity Dump is expected to fail")
+      except AssertionError as e:
+         # Check if HTTP status is 403
+         self.assertEqual(e.args[0], 403)
```

Comment:
Here also a small problem from my previous suggestion to add the fail(): the fail() will raise an AssertError with args[0] = the string of the failure, so we will fail line 142 due to line 139.

I suggest to rewrite this part as:
```python
    try:
      self.assertTlsHandshakeFailure(client_cert=config['sasCert'],
                                     client_key=config['sasKey'])
    except AssertionError:
      trigger_succeed = False
      try:
         self.TriggerFullActivityDumpAndWaitUntilComplete(config['sasCert'], config['sasKey'])
         trigger_succeed = True
      except AssertionError as e:
         # Check if HTTP status is 403
         self.assertEqual(e.args[0], 403)
      self.assertFalse(trigger_succeed, "Full Activity Dump is expected to fail")

```

Reply:
rewrote as suggested. Thanks.